opm/autodiff/WellStateFullyImplicitBlackoilSolvent.hpp

```diff
@@ -0,0 +1,41 @@
+/*
+  Copyright 2015 IRIS AS
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_WELLSTATEFULLYIMPLICITBLACKOILSOLVENT_HEADER_INCLUDED
+#define OPM_WELLSTATEFULLYIMPLICITBLACKOILSOLVENT_HEADER_INCLUDED
+
+#include <opm/autodiff/WellStateFullyImplicitBlackoil.hpp>
+
+namespace Opm
+{
+
+    class WellStateFullyImplicitBlackoilSolvent : public WellStateFullyImplicitBlackoil
```

Comment:
Do you want me to move whats in handleAdditionalWellInflow in SimulatorFullyImplicitBlackoilSolvent_impl.hpp to the wellState?
Have can I make sure WellStateFullyImplicitBlackOilSolvent's init() is called when the initialization is done in the SImulatorBase_impl()?